### ISSUE TITLE
Put the wallpaper to the local wallpapers directory on GNOME.

### ISSUE THREAD

**Issue by ChiDal** [Kharacternyk/pacwall#10]
If the image is deleted (during cleanup), the bg changes to black

**Comment by Kharacternyk** [comment]
Why the heck does it work like this? I really appreciate your field testing on GNOME.

**Comment by ChiDal** [comment]
One solution can be writing the wall to `/usr/share/backgrounds/pacwall/pacwall.png`
(and overwriting it every time the script is run), then changing the wall using gsettings

**Comment by Kharacternyk** [comment]
`/usr/share/backgrounds` doesn't have the right permissions, we would need to ask for root privileges. Does GNOME have a user-local wallpapers directory?

An obvious quick fix is to have the wallpaper in `/tmp` or `~/.cache`.

**Comment by Kharacternyk** [comment]
#3 seems to resolve the issue by passing to `gsettings` the final destination of the wallpaper. Could you test it?

**Comment by ChiDal** [comment]
> `/usr/share/backgrounds` doesn't have the right permissions, we would need to ask for root privileges. Does GNOME have a user-local wallpapers directory?
> 
> An obvious quick fix is to have the wallpaper in `/tmp` or `~/.cache`.

Using 'add picture' from the gui puts it in `$HOME/Pictures/Wallpapers` so ig that's it

**Comment by ChiDal** [comment]
> #3 seems to resolve the issue by passing to `gsettings` the final destination of the wallpaper. Could you test it?

yep it works but i have another idea:
put the wall in `$HOME/Pictures/Wallpapers` and name it pacwall-_colour_.png which allows one to have multiple colour versions to use with multiple themes

**Comment by ChiDal** [comment]
Sorry for messing up again, there is no user local directory as such. (Putting files manually in `$Home/Pictures/Wallpapers` does nothing, it has to be done via GUI).
Instead, I found [this](https://unix.stackexchange.com/a/134146).
Which means you can put it anywhere you want, you just need to make an xml.
Here's what I did:
`~/.local/share/gnome-background-properties/pacwall.xml`:
```<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE wallpapers SYSTEM "gnome-wp-list.dtd">
<wallpapers>
	<wallpaper deleted="false">
		<name>pacwall</name>
		<filename>/home/chinmay/pacwall/pacwall.png</filename>
	</wallpaper>
</wallpapers>```

Unfortunately in filename, `$HOME` or `~` doesn't work, so you need to get the username using the script.

I tested this and it works.

Sorry for bothering you for GNOME support

**Comment by Kharacternyk** [comment]
I'm planning to port the script to Debian-based distros including Ubuntu, so GNOME support is must-have.

**Comment by ChiDal** [comment]
See #13

**Comment by ChiDal** [comment]
Resolved by #13

**Comment by ChiDal** [comment]
Resolved by #13
